Missing pthread flag

[100%] Linking CXX executable utilmm_testsuite
    ../src/libutilmm.so.1.1: error: undefined reference to 'pthread_mutexattr_init'
    ../src/libutilmm.so.1.1: error: undefined reference to 'pthread_mutexattr_settype'
    ../src/libutilmm.so.1.1: error: undefined reference to 'pthread_mutexattr_destroy'
    collect2: error: ld returned 1 exit status

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,15 +4,6 @@ project(Util--)
 SET(PROJECT_VERSION 1.1)
 SET(API_VERSION 1)
 
-
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    # using Clang
-    set( CMAKE_CXX_FLAGS " -lpthread ")
-elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    # using GCC
-    set( CMAKE_CXX_FLAGS " -pthread ")
-endif()
-
 set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 FIND_PACKAGE(Boost COMPONENTS unit_test_framework)
 FIND_PACKAGE(Boost REQUIRED graph filesystem thread regex system)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,15 @@ project(Util--)
 SET(PROJECT_VERSION 1.1)
 SET(API_VERSION 1)
 
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    # using Clang
+    set( CMAKE_CXX_FLAGS " -lpthread ")
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # using GCC
+    set( CMAKE_CXX_FLAGS " -pthread ")
+endif()
+
 set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 FIND_PACKAGE(Boost COMPONENTS unit_test_framework)
 FIND_PACKAGE(Boost REQUIRED graph filesystem thread regex system)
@@ -18,9 +27,9 @@ TEST_BIG_ENDIAN(WORDS_BIGENDIAN)
 CONFIGURE_FILE(utilmm/config/config.h.in utilmm/config/config.h)
 CONFIGURE_FILE(utilmm.pc.in utilmm.pc @ONLY)
 
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/utilmm.pc 
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/utilmm.pc
     DESTINATION lib/pkgconfig)
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/utilmm/config/config.h 
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/utilmm/config/config.h
     DESTINATION include/utilmm/config)
 INSTALL(DIRECTORY utilmm/ DESTINATION include/utilmm
     PATTERN "*.in" EXCLUDE)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -26,6 +26,10 @@ set_target_properties(utilmm PROPERTIES
     SOVERSION ${API_VERSION}
     INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib)
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+target_link_libraries(utilmm Threads::Threads)
+
 target_link_libraries(utilmm ${Boost_REGEX_LIBRARY})
 target_link_libraries(utilmm ${Boost_THREAD_LIBRARY})
 target_link_libraries(utilmm ${Boost_FILESYSTEM_LIBRARY})

